client/src/pages/RootWorkspacePage/AdminControls/index.tsx

```diff
@@ -158,6 +170,68 @@ class AdminControlsPresentational extends React.Component<any, any> {
             value={this.state.schedulingPriority}
           />
         </div>
+        <div
+          style={{
+            marginBottom: "25px",
+            marginTop: "25px",
+            opacity: this.state.isDepthLimitUpdatePending ? 0.5 : 1,
+          }}
+        >
+          Depth Limit
+          <div style={{ fontWeight: 600, fontSize: "12px" }}>
+            To change, edit and press enter
+          </div>
+          <div
+            style={{
+              color: "red",
+              display: isNaN(this.state.depthLimit) ? "block" : "none",
+              fontWeight: 600,
+              fontSize: "12px",
+            }}
+          >
+            Error: non-numeric value entered
+          </div>
+          <FormControl
+            onChange={e => {
+              const target = e.target as HTMLInputElement;
+
+              this.setState({
+                depthLimit: target.value,
```

Comment:
Why not set this as a Number instead of wrapping later values?

Reply:
I tried that initially but then if the user inputs a non-numeric value into the field, the UI breaks. As a controlled input, we want this bit of state to be able to equal any possible value (and then warn the user if they've entered a non-numeric value and give them a chance to correct it). 

If you can get a less awkward approach to work, let me know. Though probably not high priority right now.

Reply:
I think a less awkward approach would be to restrict the user's inputs to valid ones instead of throwing an error message. This also prevents NaN cases, simplifying the logic. Here is a regex I found (but did not test) on SO.
```
   onChange(e){
      const target = e.target as HTMLInputElement;
      const regex = /^[0-9\b]+$/;
      if (target.value === '' || regex.test(target.value)) {
         this.setState({ depthLimit: Number(target.value)})
      }
   }
```
Another thing we need to be careful with is decimals, as depthLimit should always be an Int.

Reply:
The only really important thing for me though is to make sure we only accept `Int`, to prevent a graphql error. Did you try writing `2.4` or something as a depthLimit?